src/pm.c

```diff
@@ -0,0 +1,472 @@
+#include <stdarg.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+#define PM_COUNTER	"counter"
+#define PM_GAUGE	"gauge"
+#define PM_HISTOGRAM	"histogram"
+
+#define PM_LEN			128
+#define PM_LABEL_MAX		5
+#define PM_LABEL_VAL_MAX	5
+#define PM_METRIC_MAX		30
+#define PM_HIST_BIN_MAX		10
+
+typedef struct {
+	char		name[PM_LEN];
+	char		help[PM_LEN];
+	char		type[PM_LEN];
+	char		label[PM_LABEL_MAX][PM_LEN];
+	uint32_t	n_label;
+	double		bin[PM_HIST_BIN_MAX];
+	uint32_t	n_bin;
```

Comment:
is this just the total bucket number?

Reply:
yes, bin and buckets are synonym : https://en.wikipedia.org/wiki/Histogram
decided to go with bin in the code, since it's shorter name